controller_manager:
  ros__parameters:
    update_rate: 600  # Hz

    forward_command_controller_position:
      type: forward_command_controller/ForwardCommandController

    joint_state_controller:
      type: joint_state_controller/JointStateController

    ur_joint_trajectory_controller:
      type: joint_trajectory_controller/JointTrajectoryController

    force_torque_state_controller:
      type: ur_controllers/ForceTorqueStateController

ur_joint_trajectory_controller:
  ros__parameters:
    joints:
      - shoulder_pan_joint
      - shoulder_lift_joint
      - elbow_joint
      - wrist_1_joint
      - wrist_2_joint
      - wrist_3_joint
    state_publish_rate: 100.0
    action_monitor_rate: 20.0
    allow_partial_joints_goal: false
    constraints:
      stopped_velocity_tolerance: 0.01
      goal_time: 0.0
      shoulder_pan_joint:
        trajectory: 0.05
        goal:       0.03
      shoulder_lift_joint:
        trajectory: 0.05
        goal:       0.03
      elbow_joint:
        trajectory: 0.05
        goal:       0.03
      wrist_1_joint:
        trajectory: 0.05
        goal:       0.03
      wrist_2_joint:
        trajectory: 0.05
        goal:       0.03
      wrist_3_joint:
        trajectory: 0.05
        goal:       0.03

forward_command_controller_position:
  ros__parameters:
    joints:
      - shoulder_pan_joint
      - shoulder_lift_joint
      - elbow_joint
      - wrist_1_joint
      - wrist_2_joint
      - wrist_3_joint
    interface_name: position

force_torque_state_controller:
  ros__parameters:
    sensor_name: tcp_fts_sensor
    state_interface_names:
      - fx
      - fy
      - fz
      - tx
      - ty
      - tz
    frame_id: tool0
    topic_name: ft_data
